apps/crl.c

```diff
@@ -232,27 +233,37 @@ int crl_main(int argc, char **argv)
             goto end;
         }
 
-        xobj = X509_STORE_CTX_get_obj_by_subject(ctx, X509_LU_X509,
-                                                 X509_CRL_get_issuer(x));
-        if (xobj == NULL) {
-            BIO_printf(bio_err, "Error getting CRL issuer certificate\n");
+        matching_objs = X509_STORE_CTX_get1_objs_by_auth(ctx,
+                                                         X509_CRL_get_issuer(x),
+                                                         X509_CRL_get0_authority_key_id(x));
+        num_objs = sk_X509_OBJECT_num(matching_objs);
+        if (num_objs == 0) {
+            BIO_printf(bio_err, "verify failure\n");
             goto end;
         }
-        pkey = X509_get_pubkey(X509_OBJECT_get0_X509(xobj));
-        X509_OBJECT_free(xobj);
-        if (pkey == NULL) {
-            BIO_printf(bio_err, "Error getting CRL issuer public key\n");
-            goto end;
+        for (i = 0; i < num_objs; i++) {
```

Comment:
The subject name + AKID is supposed to uniquely identify an issuer CA, so it is unclear why you want to test them all. It should be sufficient to test only the first. The key should be identical for all CAs with the same subject name and AKID (subject key identifier field).